app/src/main/java/org/jellyfin/mobile/utils/QualityOptions.kt

```diff
@@ -0,0 +1,180 @@
+// Taken and adapted from https://github.com/jellyfin/jellyfin-web/blob/ebb4b050817d4d1920ec28209dee0539410c06ef/src/components/qualityOptions.js
+
+package org.jellyfin.mobile.utils
+
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.text.DecimalFormat
+import java.text.StringCharacterIterator
+
+object QualityOptions {
+    fun getVideoQualityOptions(options: String): String {
+        with(JSONObject(options)) {
+            return getVideoQualityOptions(
+                currentMaxBitrate = optInt("currentMaxBitrate"),
+                isAutomaticBitrateEnabled = optBoolean("isAutomaticBitrateEnabled"),
+                videoWidth = optInt("videoWidth", 4096),
+                videoHeight = optInt("videoHeight"),
+                enableAuto = optBoolean("enableAuto")
+            ).toJSONString()
+        }
+    }
+
+    private fun getVideoQualityOptions(
+        currentMaxBitrate: Int = 0,
+        isAutomaticBitrateEnabled: Boolean = false,
+        videoWidth: Int = 4096,
+        videoHeight: Int = 0,
+        enableAuto: Boolean = false
```

Comment:
The defaults don't seem to be needed here (and at other locations in this file). I'd remove them, as they needlessly increase the method count.

Reply:
Same as above. these are the JS defaults, if you think they will not be necessary I can remove them.

Reply:
I see, I think it'd be enough to add the fallbacks in the calling code, they're not necessary imo to be included within the function header itself.